pkg/virt-controller/watch/vm.go

```diff
@@ -217,6 +220,30 @@ func (c *VMController) execute(key string) error {
 		}
 	}
 
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.VolumeSource.PersistentVolumeClaim != nil {
+			key := fmt.Sprintf("%s/%s", vm.Namespace, volume.VolumeSource.PersistentVolumeClaim.ClaimName)
+			obj, exists, err := c.pvcInformer.GetStore().GetByKey(key)
+			if exists && err == nil {
+				pvc := obj.(*k8score.PersistentVolumeClaim)
+				for _, or := range pvc.ObjectMeta.OwnerReferences {
+					if or.Kind == "DataVolume" {
+						c.recorder.Eventf(vm,
```

Comment:
Here we are already at the fifth level of indentation. This reduces the readability as the reader needs to keep up with all the `and` conditions that need to be true at this level (and the scope of the nested `for` loop also).

To get rid of the nesting and in order to make the flow clearer you could use some variant of the guard clause, i.e.

```
if volume.VolumeSource.PersistentVolumeClaim == nil {
    continue
}
```

And the like.

This would produce something like this:

```
	for _, volume := range vm.Spec.Template.Spec.Volumes {
		if volume.VolumeSource.PersistentVolumeClaim == nil {
			continue
		}
		
		key := fmt.Sprintf("%s/%s", vm.Namespace, volume.VolumeSource.PersistentVolumeClaim.ClaimName)
		obj, exists, err := c.pvcInformer.GetStore().GetByKey(key)
		if err != nil {
			// TODO: handle error!
		}
		if !exists {
			continue
		}
		
		pvc := obj.(*k8score.PersistentVolumeClaim)
		for _, or := range pvc.ObjectMeta.OwnerReferences {
			if or.Kind != "DataVolume" {
				continue
			}

			c.recorder.Eventf(vm,
				k8score.EventTypeWarning,
				PVCVolumeSourceMisusedReason,
				"PVC used as volume source where Data Volume should be used %s: %v",
				volume.VolumeSource.PersistentVolumeClaim.ClaimName,
				err)
			err := fmt.Errorf("PVC %v owned by DataVolume %v cannot be used as a volume source. Use DataVolume instead",
				key, or.Name)
			logger.Reason(err).Error("Invalid VM spec")
			return err
		}
	}

```

Reply:
yup, this is more consistent with the style elsewhere in our code.